framework/Samplers/MCMC/MCMC.py

```diff
@@ -327,7 +357,61 @@ def localFinalizeActualSampling(self, jobObject, model, myInput):
       @ In, myInput, list, the generating input
       @ Out, None
     """
+    self._localReady = True
     AdaptiveSampler.localFinalizeActualSampling(self, jobObject, model, myInput)
+    prefix = jobObject.getMetadata()['prefix']
+    full = self._targetEvaluation.realization(index=self.counter-1)
+    rlz = dict((var, full[var]) for var in (list(self.toBeCalibrated.keys()) + [self._likelihood] + list(self.dependentSample.keys())))
+    rlz['traceID'] = self.counter
+    rlz['LogPosterior'] = self.inputInfo['LogPosterior']
+    rlz['AcceptRate'] = self.inputInfo['AcceptRate']
+    if self.counter == 1:
+      self._addToSolutionExport(rlz)
+      self._currentRlz = rlz
+    if self.counter > 1:
+      alpha = self._useRealization(rlz, self._currentRlz)
+      self.netLogPosterior = alpha
+      self._accepted = self._checkAcceptance(alpha)
+      if self._accepted:
+        self._currentRlz = rlz
+        self._addToSolutionExport(rlz)
+        self._updateValues = dict((var, rlz[var]) for var in self._updateValues)
+      else:
+        self._currentRlz.update({'traceID':self.counter, 'LogPosterior': self.inputInfo['LogPosterior'], 'AcceptRate':self.inputInfo['AcceptRate']})
+        self._addToSolutionExport(self._currentRlz)
+        self._updateValues = dict((var, self._currentRlz[var]) for var in self._updateValues)
+    if self._tune:
+      self._acceptInTune = self._acceptInTune + 1 if self._accepted else self._acceptInTune
+      self._countsUntilTune -= 1
+    ## tune scaling parameter
+    if not self._countsUntilTune and self._tune:
+      ### tune
+      self._scaling = self.tuneScalingParam(self._scaling, self._acceptInTune/float(self._tuneInterval))
+      ### reset counter
+      self._countsUntilTune = self._tuneInterval
+      self._acceptInTune = 0
+
+  @abc.abstractmethod
+  def _useRealization(self, newRlz, currentRlz):
```

Comment:
Do you think you need to "overload" this method to make this abstract  method? I see why you want to do it to force the derived classses to have this method but is it really required?

Reply:
I think it is needed.

Reply:
why?